Core/Src/main.c

```diff
@@ -95,10 +103,24 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	  HAL_GPIO_WritePin(GPIOB, GPIO_PIN_8, GPIO_PIN_RESET); //setting the PB4 port (I think that's the GPIO port?) low
+
+	  HAL_SPI_TransmitReceive(&hspi1, T_buffer, R_buffer, 8, 500);
```

Comment:
For `HAL_SPITransmitReceive` you are only transmitting 3 bytes so change 8 to 3!